api/v1beta1/tempest_types.go

```diff
@@ -38,22 +38,162 @@ type Hash struct {
 // TempestSpec TempestRun parts
 type TempestRunSpec struct {
 	// +kubebuilder:validation:Optional
-	// +kubebuilder:default={"tempest.api.identity.v3"}
-	// AllowedTests
-	AllowedTests []string `json:"allowedTests,omitempty"`
+	// +kubebuilder:default="tempest.api.identity.v3"
+	// IncludeList
+	IncludeList string `json:"includeList,omitempty"`
 
 	// +kubebuilder:validation:Optional
-	// SkippedTests
-	SkippedTests []string `json:"skippedTests,omitempty"`
+	// +kubebuilder:default=""
+	// ExcludeList
+	ExcludeList string `json:"excludeList,omitempty"`
 
-        // +kubebuilder:validation:Optional
-	// +kubebuilder:default:=0
-        // Concurrency is the Default concurrency
-        Concurrency *int64 `json:"concurrency,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=-1
+	// Concurrency is the Default concurrency
+	Concurrency int64 `json:"concurrency,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	// Smoke tests
+	Smoke bool `json:"smoke,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=true
+	// Run tests in parallel
+	Parallel bool `json:"parallel,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	// Serial run
+	Serial bool `json:"serial,omitempty"`
```

Comment:
aren't serial and Parallel mutually exclusive? Can it be that both will be set to "True" for example?

Reply:
It is a good point. I was thinking about the same. But I'm not sure how much should the test-operator contain the logic of using the parameters. For me it is fine that the pod fails when the user decides to use the parameters incorrectly. But I guess it is up for a discussion (@kopecmartin, @afazekas) (?)

Reply:
if a user passes both, --serial and --parallel, arguments, tempest will throw an error:
`tempest run: error: argument --parallel: not allowed with argument --serial/-t`

The way I see it, we don't need to check any argument compatibility on the operator/tempest image side - that's done by tempest itself. The operator together with the image definitions are responsible for getting the arguments from user to tempest - they are just a messenger.